### PR TITLE
147 render notebooks into sphinx documentation

### PR DESCRIPTION
**Pull request by bonh**
second try using docker.

### CODE REVIEW

**Comment by bonh** [comment]
That seems to work; a Docker image is built from `MParT-examples` on push. This image contains all the notebooks executed via `nbconvert`. On push to `MParT` the action is executed to build the documentation (as before). This action pulls the Docker image, extracts the notebook files, and renders the notebooks via `nbsphinx`. `nbsphinx` is not used to execute the notebooks because that takes forever in a GitHub action (don't know why).

To render more examples you have to
1. add the example to `MParT-examples`,
2. add the filename to `jupytext` and `nbconvert` in `MPart-examples/.binder/postBuild`,
3. push to main (or pull request, sure),
4. wait until the Docker image is build from `MPart-examples`,
5. and add the example name to `docs/source/tutorials/index.rst`.

Note, that I haven't looked at the actual *.html files yet as the push to gh pages is still commented out!

File: .github/workflows/build-doc.yml [hunk 1]
```diff
@@ -3,41 +3,53 @@ name: docs
 on:
   push:
     branches:
-      - main
+      - 147-render-notebooks-into-sphinx-documentation
```
Comment (dannys4) [inline .github/workflows/build-doc.yml:6]: Just don't forget to change this back before merging

File: .github/workflows/build-doc.yml [hunk 2]
```diff
@@ -3,41 +3,53 @@ name: docs
...
+        shell: bash -l {0}
+        run: |
+           cd ${{ github.workspace }} && cd build
+           make sphinx
+
+      #- name: Push to gh-pages
+      #  uses: peaceiris/actions-gh-pages@v3.6.1
+      #  with:
+      #    github_token: ${{ secrets.GITHUB_TOKEN }}
+      #    publish_dir: build/docs/sphinx/
```
Comment (dannys4) [inline .github/workflows/build-doc.yml:54]: I think we should push to gh-pages once before merging (i.e. before changing the `branch: 147-....` back to `branch: main` from above, we should uncomment this and see if it gives the docs we expect)

Comment (bonh) [reply .github/workflows/build-doc.yml]: Will do!

Comment (bonh) [reply .github/workflows/build-doc.yml]: That didn't go well: https://measuretransport.github.io/MParT/source/tutorials/MonotoneLeastSquares.html

Comment (bonh) [reply .github/workflows/build-doc.yml]: Juhuuuuu: https://measuretransport.github.io/MParT/source/tutorials/MonotoneLeastSquares.nbconvert.html

**Review by dannys4 — APPROVED** [review]

**Pull request merged**